internal/featuretests/v3/globalratelimit_test.go

```diff
@@ -677,6 +808,28 @@ func TestGlobalRateLimiting(t *testing.T) {
 								Name:      "fallback-cert",
 								Namespace: "default",
 							}
+
+							httpProxyProcessor.GlobalRateLimitService = &contour_api_v1alpha1.RateLimitServiceConfig{
+								ExtensionService: contour_api_v1alpha1.NamespacedName{
+									Name:      "extension",
+									Namespace: "ratelimit",
+								},
+								Domain: "contour",
+								DefaultGlobalRateLimitPolicy: &contour_api_v1.GlobalRateLimitPolicy{
+									Descriptors: []contour_api_v1.RateLimitDescriptor{
```

Comment:
a drawback of adding to this test this way is we have to set "Disabled" on the HTTPProxies in all the cases where we don't expect a rate limit to be set, not a huge thing but makes the test coverage a little different

Reply:
Yes, unfortunately, the way these tests are written doesn't allow multiple global rate limit services to be configured easily. All HTTPProxy uses the same RLS and that is why the opted-in by default.